### PR TITLE
Update CI Workflow

### PR DESCRIPTION
**Pull request by silvia-ivanova-github-rep**
**Description**

- A CI workflow that includes release.
- Changed trigger events (_pull request_ to _push_)
- The workflow will be executed when a push to the main branch occurs.
- This workflow is already tested in a test repository and it works.

**Affects**
All team members and the entire workflow.

**Notes for Reviewer**
Please review & comment/approve.

### CODE REVIEW

File: .github/workflows/android-release.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      #command: run unit tests
+      - name: Run Build Test
+        #Working directory is the folder with the project
+        working-directory: ./BibBuddy
+        run: bash ./gradlew test --stacktrace
```
Comment (claudia3) [inline .github/workflows/android-release.yml:29]: We already run the unit tests in the android.yml workflow when we create a PR to dev. So I don't think it is necessary to run the unit tests again.

File: .github/workflows/android-release.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+      #command:  Run Java Checkstyle
+      - name: Run Java Checkstyle
```
Comment (claudia3) [inline .github/workflows/android-release.yml:32]: We already run checkstyle in the android.yml workflow when we create a PR to dev. Reason see above.

File: .github/workflows/android-release.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+name: APK Workflow
+
+#Event triggers
+#execute the jobs everytime when a pull_request to the main-branch is created
+on:
+  push:
+    branches: main
+
+jobs:
+  runtests:
```
Comment (claudia3) [inline .github/workflows/android-release.yml:11]: Do we really need to run the tests again. Because in dev they were already run. See comment.

Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:11]: This part includes java checkstyle. As we tested before, there are checkstyle errors that we should resolve.

File: .github/workflows/android-release.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1.1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+        with:
+          #The name of the tag for this release
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          body: This is a release of the created APK
```
Comment (claudia3) [inline .github/workflows/android-release.yml:83]: The release is created but the created release is not uploaded yet?

File: .github/workflows/android-release.yml [hunk 5]
```diff
@@ -0,0 +1,83 @@
+#This is a workflow that creates, downloads, uploads and releases APK-file.
```
Comment (claudia3) [inline .github/workflows/android-release.yml:1]:
```suggestion
# This is a workflow that creates, downloads, uploads and releases APK-file.
```

File: .github/workflows/android-release.yml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        with:
+          checkstyle_config: google_checks.xml
+          tool_name: Java Checkstyle
+
+  apk:
+    name: APK
+
+    #to run this job, the job build_test should be successful  
+    #otherwise: this job will be skipped
+    needs: runtests
```
Comment (claudia3) [inline .github/workflows/android-release.yml:45]: Do we really need runtests? Reason see other comment.
```suggestion

```

File: .github/workflows/android-release.yml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Setup Java JDK
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Build APK
+        working-directory: ./BibBuddy
+        run: bash ./gradlew assembleDebug --stacktrace
+
+      #Upload the APK as a build artifact that can be used by subsequent workflow steps
```
Comment (claudia3) [inline .github/workflows/android-release.yml:60]:
```suggestion
      # Upload the APK as a build artifact that can be used by subsequent workflow steps
```

File: .github/workflows/android-release.yml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Upload APK
+        uses: actions/upload-artifact@v2.2.2
+        with:
+          name: apk_file
+          path: ./BibBuddy/app/build/outputs/apk/debug/app-debug.apk 
+
+      #Download a build artifact that was previously uploaded in the workflow by the upload-artifact action
```
Comment (claudia3) [inline .github/workflows/android-release.yml:67]:
```suggestion
      # Download a build artifact that was previously uploaded in the workflow by the upload-artifact action
```

File: .github/workflows/android-release.yml [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Download APK
+        uses: actions/download-artifact@v2.0.8
+        with:
+          name: apk_file
+
+      #Create a release for a tag in your repository
```
Comment (claudia3) [inline .github/workflows/android-release.yml:73]:
```suggestion
      # Create a release for a tag in your repository           
```

File: .github/workflows/android-release.yml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+          name: apk_file
+
+      #Create a release for a tag in your repository           
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1.1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+        with:
+          #The name of the tag for this release
```
Comment (claudia3) [inline .github/workflows/android-release.yml:80]:
```suggestion
          # The name of the tag for this release
```

**Review by claudia3 — CHANGES_REQUESTED** [review]
Please have a look at my comments. Please add a space after a comment.

File: .github/workflows/android-release.yml (hunk 4 above)
Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:83]: The release is uploaded as a part of the artifact. Everyone can download the created apk:
image

See this part of the .yml

```
      #Upload the APK as a build artifact that can be used by subsequent workflow steps
      - name: Upload APK
        uses: actions/upload-artifact@v2.2.2
        with:
          name: apk_file
          path: ./BibBuddy/app/build/outputs/apk/debug/app-debug.apk 

```

File: .github/workflows/android-release.yml (hunk 2 above)
Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:32]: Yes, but as I made a test from my currently created branch to main-branch, there were some checkstyle errors.  see #92 

Before our CI workflow integration, we merged code to dev, which is not already checked.

Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:32]: Here is the workflow run:
https://github.com/UniRegensburg/unsere-app-fur-die-universitat-regensburg-bib-buddy/actions/runs/607611443

File: .github/workflows/android-release.yml (hunk 1 above)
Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:29]: Are there unit tests that are pushed to dev, before our workflow integration? The workflow checks the unit tests from the other branches when a pull request to dev is made. So, our workflow now does not check the unit tests that are merged to dev before our CI workflow integration.

File: .github/workflows/android-release.yml (hunk 4 above)
Comment (claudia3) [reply .github/workflows/android-release.yml:83]: Ok, should the APK also be released in Google Play Store with our bibBuddy account?

File: .github/workflows/android-release.yml (hunk 1 above)
Comment (claudia3) [reply .github/workflows/android-release.yml:29]: There were example unit tests before our workflow integration.

File: .github/workflows/android-release.yml (hunk 4 above)
Comment (silvia-ivanova-github-rep) [reply .github/workflows/android-release.yml:83]: yes, I think so.

**Comment by claudia3** [comment]
As discussed in a meeting the tests will be left for testing at the moment.

**Review by claudia3 — APPROVED** [review]

**Pull request merged**